packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -156,10 +152,12 @@ export class HostedPluginSupport {
         this.theiaReadyPromise = Promise.all([this.preferenceServiceImpl.ready, this.workspaceService.roots]);
         this.workspaceService.onWorkspaceChanged(() => this.updateStoragePath());
 
-        for (const id of this.monacoTextmateService.activatedLanguages) {
-            this.activateByLanguage(id);
+        const modeService = monaco.services.StaticServices.modeService.get();
+        for (const modeId of Object.keys(modeService['_instantiatedModes'])) {
```

Comment:
We are planning to add integration tests.

Another idea I had is to use monaco via esm and generated d.ts files for js esm files. It would be easier than do manual checking.

Reply:
discussed, that in this case it will fail at runtime anyway